sources/sql/src/producer.rs

```diff
@@ -486,17 +493,22 @@ pub fn binary_op_to_sql(lhs: SQLExpr, rhs: SQLExpr, op: ast::BinaryOperator) ->
     }
 }
 
-fn new_table_alias(alias: String) -> ast::TableAlias {
+fn new_table_alias(alias: String, dialect: &str) -> ast::TableAlias {
     ast::TableAlias {
-        name: new_ident(alias),
+        name: new_ident(alias, dialect),
         columns: Vec::new(),
     }
 }
 
-fn new_ident(str: String) -> ast::Ident {
+fn new_ident(str: String, dialect: &str) -> ast::Ident {
```

Comment:
We may want to refactor this to methods on a struct to more easily pass serialization hints around.

Reply:
Thought about this - perhaps for a future refactor PR?